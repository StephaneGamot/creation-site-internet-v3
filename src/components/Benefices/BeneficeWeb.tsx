"use client";

import { motion } from "framer-motion";
import { useTranslations } from "next-intl";

type BenefitItem = {
  id: number;
  title: string;
  desc: string;
};

export default function BeneficeWeb() {
  const t = useTranslations("beneficeWeb");
  const items: BenefitItem[] = t.raw("list");

  return (
    <section className="max-w-5xl mx-auto text-center">
      <h2 className="text-3xl font-title font-semibold mb-8">{t("title")}</h2>
      <div className="grid gap-8 md:grid-cols-3 text-left">
        {items.map((item, index) => (
          <motion.div
            key={item.id ?? index}
            initial={{ opacity: 0, y: 20 }}
            whileInView={{ opacity: 1, y: 0 }}
            transition={{ delay: index * 0.2 }}
            className="bg-[#f9f9f9] p-6 rounded-2xl shadow"
          >
            <h3 className="text-xl font-semibold mb-2">{item.title}</h3>
            <p className="text-sm text-gray-600">{item.desc}</p>
          </motion.div>
        ))}
      </div>
    </section>
  );
}
