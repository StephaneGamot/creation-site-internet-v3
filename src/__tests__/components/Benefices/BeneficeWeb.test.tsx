import React, { forwardRef } from "react";
import { render, screen } from "@testing-library/react";
import BeneficeWeb from "@/components/Benefices/BeneficeWeb";
import renderer from "react-test-renderer";

// ✅ Mock de next-intl (t et t.raw)
jest.mock("next-intl", () => ({
  useTranslations: () => {
    const t = (key: string) => {
      if (key === "title") return "Les bénéfices d’un site bien conçu";
      return "";
    };

    t.raw = (key: string) => {
      if (key === "list") {
        return [
          { id: 1, title: "Rapide", desc: "Chargement éclair pour vos visiteurs." },
          { id: 2, title: "Optimisé", desc: "Structure SEO-friendly dès le départ." },
        ];
      }
      return [];
    };

    return t;
  },
}));

// ✅ Mock framer-motion typé
jest.mock("framer-motion", () => ({
  motion: {
    div: forwardRef<HTMLDivElement, any>((props, ref) => (
      <div ref={ref} {...props} />
    )),
  },
}));

// ✅ Tests
describe("BeneficeWeb", () => {
  it("rend le titre et les bénéfices", () => {
    render(<BeneficeWeb />);

    // Titre principal
    expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent("Les bénéfices d’un site bien conçu");

    // Bénéfices (titres et descriptions)
    expect(screen.getByText("Rapide")).toBeInTheDocument();
    expect(screen.getByText("Chargement éclair pour vos visiteurs.")).toBeInTheDocument();
    expect(screen.getByText("Optimisé")).toBeInTheDocument();
    expect(screen.getByText("Structure SEO-friendly dès le départ.")).toBeInTheDocument();

    // Nombre de blocs (heading level 3)
    expect(screen.getAllByRole("heading", { level: 3 })).toHaveLength(2);
  });

  it("correspond au snapshot", () => {
  const { container } = render(<BeneficeWeb />);
  expect(container).toMatchSnapshot();
});

});
