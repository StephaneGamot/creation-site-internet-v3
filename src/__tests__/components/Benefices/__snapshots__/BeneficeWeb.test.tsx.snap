// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BeneficeWeb correspond au snapshot 1`] = `
<div>
  <section
    class="max-w-5xl mx-auto text-center"
  >
    <h2
      class="text-3xl font-title font-semibold mb-8"
    >
      Les bénéfices d’un site bien conçu
    </h2>
    <div
      class="grid gap-8 md:grid-cols-3 text-left"
    >
      <div
        class="bg-[#f9f9f9] p-6 rounded-2xl shadow"
        initial="[object Object]"
        transition="[object Object]"
        whileinview="[object Object]"
      >
        <h3
          class="text-xl font-semibold mb-2"
        >
          Rapide
        </h3>
        <p
          class="text-sm text-gray-600"
        >
          Chargement éclair pour vos visiteurs.
        </p>
      </div>
      <div
        class="bg-[#f9f9f9] p-6 rounded-2xl shadow"
        initial="[object Object]"
        transition="[object Object]"
        whileinview="[object Object]"
      >
        <h3
          class="text-xl font-semibold mb-2"
        >
          Optimisé
        </h3>
        <p
          class="text-sm text-gray-600"
        >
          Structure SEO-friendly dès le départ.
        </p>
      </div>
    </div>
  </section>
</div>
`;
